api/src/test/java/uk/gov/dhsc/htbhf/dwp/repository/UCHouseholdRepositoryTest.java

```diff
@@ -42,4 +43,22 @@ void saveAndRetrieveHousehold() {
         assertThat(savedHousehold).isEqualToComparingFieldByFieldRecursively(household);
     }
 
-}
\ No newline at end of file
+    @Test
+    void shouldFindUCHouseholdByNino() {
+        String nino = "QQ123456A";
+        UCHousehold household = aHousehold();
+        repository.save(household);
+
+        Optional<UCHousehold> result = repository.findHouseholdByAdultWithNino(nino);
+
+        assertThat(result.get()).isEqualTo(household);
+    }
+
+    @Test
+    void shouldFailToFindUCHouseholdByNino() {
```

Comment:
Can we add a household with a different NINO?

Reply:
Done.